nodes/vyosnetworks_vyos/helpers.go

```diff
@@ -0,0 +1,157 @@
+package vyosnetworks_vyos
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/charmbracelet/log"
+	"github.com/scrapli/scrapligo/driver/network"
+	"github.com/srl-labs/containerlab/utils"
+	"github.com/steiler/acls"
+)
+
+func (n *vyos) save(_ context.Context, cli *network.Driver) error {
+	log.Debug("Saving config", "node", n.Cfg.ShortName)
+	resp, err := cli.SendConfigs(saveCmd)
+	if err != nil {
+		return err
+	} else if resp.Failed != nil {
+		return fmt.Errorf("Save failed. Response: %w", err)
+	}
+	log.Info("Save successful", "node", n.Cfg.ShortName)
+	return nil
+}
+
+func (n *vyos) newCli() (*network.Driver, error) {
+	cli, err := utils.SpawnCLIviaExec(
+		scrapliPlatformName,
+		n.Cfg.LongName,
+		n.Runtime.GetName())
+	if err != nil {
+		return nil, err
+	}
+	return cli, nil
+}
+
+func (n *vyos) createVyosFiles(_ context.Context) error {
+	nodeCfg := n.Config()
+
+	// generate config dir
+	utils.CreateDirectory(n.configDir, 0777)
+	log.Debugf("Chowning dir %s", n.configDir)
+	if err := os.Chown(n.Cfg.LabDir, 0, vyattacfg_gid); err != nil {
+		return err
+	}
+
+	nodeCfg.MgmtIntf = "eth0"
+
+	// use startup config file provided by a user
+	if nodeCfg.StartupConfig != "" {
+		c, err := os.ReadFile(nodeCfg.StartupConfig)
+		if err != nil {
+			return err
+		}
+		cfgTemplate = string(c)
+	}
+
+	err := n.GenerateConfig(nodeCfg.ResStartupConfig, cfgTemplate)
+	if err != nil {
+		return err
+	}
+
+	// Vyos needs pre and post boot config scripts to be present even if they're noops
+	scriptDir := filepath.Join(n.configDir, "scripts")
+	preScript := filepath.Join(scriptDir, "vyos-preconfig-bootup.script")
+	postScript := filepath.Join(scriptDir, "vyos-postconfig-bootup.script")
+	utils.CreateDirectory(scriptDir, 0777)
+
+	for _, s := range []string{preScript, postScript} {
+		utils.CreateFile(s, "#!/bin/sh")
+		os.Chmod(s, 0755)
+		os.Chown(s, 0, vyattacfg_gid)
+	}
+
+	return nil
+}
+
+// clab applies ACLs to the node directory based on the idea that the NOS
+// parses everything through a management system. While Vyos does have a
+// management system it does a bunch of stuff as a regular linux user so the
+// directory needs rw access for the vyattacfg group
+func (n *vyos) fixdirACL() error {
+	log.Debugf("Setting up %s ACLs", n.Cfg.LabDir)
+	a := &acls.ACL{}
+	if err := a.Load(n.Cfg.LabDir, acls.PosixACLAccess); err != nil {
+		return err
+	}
+	entry := acls.NewEntry(acls.TAG_ACL_GROUP, uint32(vyattacfg_gid), 7)
+	if err := a.AddEntry(entry); err != nil {
+		return err
+	}
+	if err := a.Apply(n.Cfg.LabDir, acls.PosixACLAccess); err != nil {
+		return err
+	}
+	if err := a.Apply(n.Cfg.LabDir, acls.PosixACLDefault); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Convert the PKCS#1 formatted key that clab generates into a PKCS#8 that Vyos
+// requires
+func pkcs1To8(der []byte) ([]byte, error) {
```

Comment:
@hellt I wonder if this is something that should go in the utils package, or maybe default to PKCS#8 and offer a util function for the reverse? PKCS#1 is falling out of favor. It's encrypted form is considered insecure, it's not the default for OpenSSL anymore, doing anything with it requires using an additional `-traditional` flag, TLS1.3 doesn't allow it. 

That's probably why it doesn't work with VyOS out of the box

Reply:
yeah I am all up for using fresher security mechanisms
Will be up for handling this in another PR